File: Factorial.py
def factorial(x):
    if(x == 1):
        return x
    else:
        return x * factorial(x - 1)
print("Work out the factorial of... \n")
n = input()
r = factorial(int(n))
print("The factorial of " + str(n) + " is " + str(r))

File: Factorial.cpp
#include<iostream>
using namespace std;

float factorial(float x) {
      return(x == 1) ? x : x * factorial(x-1);
}

int main(){
	float n;
	cout << "Work out the factorial of..." << endl;
	cin >> n;
	cout << "The factorial of " << n << " is " << factorial(n);;
}

